### PR TITLE
Make `RouteLineTrimOffset` public

### LINKED ISSUE #5851: Dokka fails for value classes
https://github.com/mapbox/mapbox-navigation-android/pull/5842 introduced a Kotlin `value class` and an attempt to use Dokka to generate the HTML docs for that class fail and produce undocumented pages, which also makes our tests fail.

Currently making the class `internal` but we might want to look for a root cause so that this doesn't block future `value class` usage.

cc @RingerJK for visibility in case you've come across this before

### LINKED ISSUE #5852: Metalava misconfigured for Kotlin value classes
https://github.com/mapbox/mapbox-navigation-android/pull/5842 introduced a Kotlin `value class` that even when marked `internal` was producing an entry in metalava's API files. For example:
```kotlin
@JvmInline
internal value class RouteLineTrimOffset(val offset: Double)
```
results in:
```
   public final class RouteLineTrimOffset {
    ctor public RouteLineTrimOffset(double offset);
    method public double getOffset();
    property public final double offset;
  }
```

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description

- ~~Moves `MapboxRouteLineUtils` from `internal` package to `util` and~~ makes `RouteLineTrimOffset` public
Fixes #5851 - fixed after bumping to latest Dokka version `1.6.21` https://github.com/mapbox/mapbox-navigation-android/blob/bd50b6c84311a360f9ec96c21c24c4488bbe9f2e/gradle/dependencies.gradle#L186
Fixes #5852 - as mentioned in https://github.com/mapbox/mapbox-navigation-android/issues/5852#issuecomment-1136176973 this is not an issue anymore after making `RouteLineTrimOffset` public

### CODE REVIEW

**Review by kmadsen — APPROVED** [review]
Considering this is exposing a new public api, it appears worthy of a changelog

Can you add something about `RouteLineTrimOffset`

**Comment by Guardiola31337** [comment]
> Considering this is exposing a new public api, it appears worthy of a changelog
>
> Can you add something about `RouteLineTrimOffset`

Yeah, that makes sense 👍  @cafesilencio could you help here? What did you have in mind?

**Comment by cafesilencio** [comment]
MapboxRouteLineUtils needs to remain internal.

**Comment by cafesilencio** [comment]
Why does the package have to change?

**Comment by cafesilencio** [comment]
This shouldn't be merged as is.  MapboxRouteLineUtils shouldn't be changing.

**Comment by Guardiola31337** [comment]
> MapboxRouteLineUtils needs to remain internal.
> Why does the package have to change?

It is already marked as `internal` https://github.com/mapbox/mapbox-navigation-android/blob/2004fe3360f1970abc00fa08467c0ff8031a915e/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/MapboxRouteLineUtils.kt#L50 so no need to place it under the `internal` package. If it is a problem I can revert that change.

**Comment by cafesilencio** [comment]
> > MapboxRouteLineUtils needs to remain internal.
> > Why does the package have to change?
> 
> It is already marked as `internal`
> 
> https://github.com/mapbox/mapbox-navigation-android/blob/2004fe3360f1970abc00fa08467c0ff8031a915e/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/MapboxRouteLineUtils.kt#L50
> 
> so no need to place it under the `internal` package. If it is a problem I can revert that change.

No changes to MapboxRouteLineUtils

**Comment by Guardiola31337** [comment]
> This shouldn't be merged as is. MapboxRouteLineUtils shouldn't be changing.

What's the issue? In any case, as above-mentioned, if it's causing confusion I can revert that change.

**Comment by cafesilencio** [comment]
> > This shouldn't be merged as is. MapboxRouteLineUtils shouldn't be changing.
> 
> What's the issue? In any case, as above-mentioned, if it's causing confusion I can revert that change.

Please no changes to MapboxRouteLineUtils.

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/RouteLineTrimOffset.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.mapbox.navigation.ui.maps.route.line
```
Comment (cafesilencio) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/RouteLineTrimOffset.kt:1]: This should be in the model package next to the other classes it's related to.

Comment (Guardiola31337) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/RouteLineTrimOffset.kt]: Addressed and re-pushed, let me know if looks good now 🙏

**Review by cafesilencio — APPROVED** [review]

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/RouteLineTrimOffset.kt (hunk 1 above)
Comment (cafesilencio) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/RouteLineTrimOffset.kt]: merge it

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -183,7 +183,7 @@ ext {
       jacoco               : '0.2',
       googleServices       : '4.3.3',
       mapboxSdkVersions    : '1.1.3',
-      dokka                : '1.4.20',
+      dokka                : '1.6.21',
```
Comment (RingerJK) [inline gradle/dependencies.gradle:186]: in previous versions I checked (~1.5.x) was changed entry point from `_modules.html` to another file. Could you check if it's true for this version? If so, we need support from the `docs team` to support these changes

Comment (Guardiola31337) [reply gradle/dependencies.gradle:186]: Going to merge here and we're going to test publishing the docs in `staging` / locally so we can confirm that everything is working fine before next week's release 🚀

Comment (RingerJK) [reply gradle/dependencies.gradle:186]: let's wait for android-docs PR to be merged 🙏 and then merge this one

**Comment by RingerJK** [comment]
@Guardiola31337 I noted that undocumented warnings are not working. You can check it if remove any documentation and run `./gradlew dokkaHtmlMultiModule`. We need to fix it in this PR

**Comment by RingerJK** [comment]
`android-docs` setup expects new Dokka since 2.6.0-beta.1 we need to wait for the next release and then merge this PR

**Review by RingerJK — APPROVED** [review]

**Pull request merged**